runHarness/RunProcedures/PrepareOnlyRunProcedure.pm

```diff
@@ -52,36 +52,49 @@ override 'initialize' => sub {
 
 sub run {
 	my ( $self ) = @_;
+	my $majorSequenceNumberFile = $self->getParamValue('sequenceNumberFile');
 	my $tmpDir             = $self->getParamValue('tmpDir');
 	my $weathervaneHome    = $self->getParamValue('weathervaneHome');
 	my $console_logger     = get_logger("Console");
 	my $debug_logger       = get_logger("Weathervane::RunProcedures::PrepareOnlyRunProcedure");
 	my @pids;
 	my $pid;
 
+	# Get the major sequence number
+	my $majorSeqNum;
+	if ( -e "$majorSequenceNumberFile" ) {
+		open SEQFILE, "<$majorSequenceNumberFile";
+		$majorSeqNum = <SEQFILE>;
+		close SEQFILE;
+		$majorSeqNum--; #already incremented in weathervane.pl
+	} else {
+		print "Major sequence number file is missing.\n";
+		exit -1;
+	}
 	# Get the minor sequence number of the next run
-	my $sequenceNumberFile = "$tmpDir/minorsequence.num";
-	my $seqnum;
-	if ( -e "$sequenceNumberFile" ) {
-		open SEQFILE, "<$sequenceNumberFile";
-		$seqnum = <SEQFILE>;
+	my $minorSequenceNumberFile = "$tmpDir/minorsequence.num";
+	my $minorSeqNum;
+	if ( -e "$minorSequenceNumberFile" ) {
+		open SEQFILE, "<$minorSequenceNumberFile";
+		$minorSeqNum = <SEQFILE>;
 		close SEQFILE;
-		open SEQFILE, ">$sequenceNumberFile";
-		my $nextSeqNum = $seqnum + 1;
+		open SEQFILE, ">$minorSequenceNumberFile";
+		my $nextSeqNum = $minorSeqNum + 1;
 		print SEQFILE $nextSeqNum;
 		close SEQFILE;
 	}
 	else {
-		$seqnum = 0;
-		open SEQFILE, ">$sequenceNumberFile";
+		$minorSeqNum = 0;
+		open SEQFILE, ">$minorSequenceNumberFile";
 		my $nextSeqNum = 1;
 		print SEQFILE $nextSeqNum;
 		close SEQFILE;
 	}
+	my $seqnum = $majorSeqNum . "." . $minorSeqNum;
```

Comment:
I approved this but on further testing I would like to request one change.  Can you concatenate the major and minor numbers with a "-" rather than a ".".  Otherwise spreadsheet programs will treat the result as a floating point number rather than a string and may truncate.  E.g. run "42.0" displays as "42", with the 0 not displayed.